packages/rum-core/src/domain/rumEventsCollection/view/trackScrollMetrics.ts

```diff
@@ -0,0 +1,40 @@
+import type { Duration } from '@datadog/browser-core'
+import { throttle, addEventListener, DOM_EVENT } from '@datadog/browser-core'
+
+export interface ScrollMetrics {
+  maxScrollDepth?: number
+  maxscrollHeight?: number
+  maxScrollDepthTimestamp?: Duration
+}
+
+export function trackScrollMetrics() {
+  const scrollMetrics: ScrollMetrics = {}
+
+  const handleScrollEvent = throttle(
+    (event: Event) => {
+      const scrollTop =
+        window.pageYOffset || (document.documentElement || document.body.parentNode || document.body).scrollTop
+
+      const scrollHeight = Math.round(document.documentElement.scrollHeight)
+      const scrollDepth = Math.round(Math.min(document.documentElement.clientHeight + scrollTop, scrollHeight))
+
+      if (scrollDepth > (scrollMetrics?.maxScrollDepth ?? 0)) {
+        scrollMetrics.maxScrollDepth = scrollDepth
+        scrollMetrics.maxscrollHeight = scrollHeight
+        scrollMetrics.maxScrollDepthTimestamp = event.timeStamp as Duration
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I wonder whether using `event.timeStamp` is a good idea here, because it is known to be buggy on some occasion. Ideally, we would use `timeStampNow()`, but since this callback is debounced, it would be a bit imprecise.